test/Serializer.js

```diff
@@ -0,0 +1,251 @@
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("isActive");
+		expect(result).to.have.property("id");
+		expect(result).to.have.property("name");
+		expect(result).to.have.property("email");
+		expect(result).to.have.property("phone");
+		expect(result).to.have.property("passwordHash");
+		expect(result).to.have.property("status");
+		expect(result).to.have.property("isActive");
+	});
+
+	it("Should add all document fields to the output prior to running both include and exclude statements", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("hideSecure");
+		expect(result).to.have.property("id");
+		expect(result).to.have.property("name");
+		expect(result).to.have.property("email");
+		expect(result).to.have.property("phone");
+		expect(result).to.have.property("status");
+		expect(result).to.not.have.property("passwordHash");
+	});
+
+	it("Should fail while serializing a document and return a blank object", () => {
+		addSerializers();
+		const docs = createDocuments();
+		const result = docs[0].serialize("nonExistingSerializer");
+		expect(Object.keys(result)).to.be.an("array").length(0);
+	});
+
+	it("Should throw errors on invalid usage", () => {
+		expect(() => {
+			User.serializer.add({invalidUsage: "Should fail"}, ["name", "email"]);
+		}).to.throw("Field name is required and should be of type string");
+
+		expect(() => {
+			User.serializer.add("broken", "invalidOptionsUsage");
+		}).to.throw("Field options is required and should be an object or array");
+
+		expect(() => {
+			User.serializeMany({notAnArray: "ofDocuments"});
+		}).to.throw("documentsArray must be an array of document objects");
+	});
+
+	const addSerializers = () => {
+		User.serializer.add("contactInfoOnly", ["name", "email", "phone"]);
+		User.serializer.add("hideSecure", {exclude: ["passwordHash"]});
+		User.serializer.add("redundant", {include: ["email", "phone", "doesntExist"], exclude: ["phone", "doesntExist"]});
+		User.serializer.add("isActiveNoStatus", {
+			exclude: ["status"],
+			modify: (serialized, original) => {
+				serialized.isActive = original.status === "active";
+				return serialized;
+			}
+		});
+		User.serializer.add("isActive", {
+			modify: (serialized, original) => {
+				serialized.isActive = original.status === "active";
+				return serialized;
+			}
+		});
+	};
+
+	const createDocuments = () => {
+		const docs = [];
+
+		docs[0] = new User({
+			id: 1,
+			name: "User One",
+			email: "userone@example.com",
+			phone: "0123456789",
+			passwordHash: "5dxFDpyKeEKiVUSp9I6dQ1DGd3CFH5Jk",
+			status: "not_active"
+		});
+
+		docs[1] = new User({
+			id: 2,
+			name: "User Two",
+			email: "usertwo@example.com",
+			phone: "0123456789",
+			passwordHash: "KnbuZWU0RNPPXhgPNMxovLtUiuMN4I6i",
+			status: "active"
+		});
+
+		return docs;
+	};
```

Comment:
I think we should move these to the top. And ideally I think these should probably be `beforeEach` hooks. And we could have a higher scoped variable called `docs`. Again, just to maintain consistency throughout. Not sure if the way I wrote the other tests is the best way, but seems beneficial to maintain consistency.